incubator/hnc/pkg/kubectl/tree.go

```diff
@@ -35,27 +35,28 @@ var (
 
 var treeCmd = &cobra.Command{
 	Use:   "tree",
-	Short: "Displays the hierarchy tree rooted at the given namespace",
-	Args:  cobra.MaximumNArgs(1),
+	Short: "Displays the hierarchy tree rooted at the given namespaces separated by space and if no namespace is provided, displays the tree starting at every root namespace on the cluster.",
 	Run: func(cmd *cobra.Command, args []string) {
+		defaultList := len(args) == 0
 		footnotesByMsg = map[string]int{}
 		footnotes = []string{}
-		nsList := make([]string, 0)
-		if len(args) != 0 {
-			nsList = append(nsList, args[0])
+		var nsList []string
+		if defaultList {
+			nsList = getAllNamespaces()
 		} else {
-			allNamespaces := getAllNamespaces()
-			nsList = append(nsList, allNamespaces...)
+			nsList = args
 		}
 		for _, nnm := range nsList {
 			hier := getHierarchy(nnm)
-			if (hier.Spec.Parent == "" && len(args) == 0) || len(args) != 0 {
-				fmt.Println(nameAndFootnotes(hier))
-				printSubtree("", hier)
+			if defaultList && hier.Spec.Parent != "" {
+				continue
```

Comment:
Maybe a comment here is worthwhile: "If we're showing the default list, skip all non-root namespaces since they'll be displayed as part of another namespace's tree."